cli/xml-to-ts.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from "fs";
+import { pascalCase } from "change-case";
+import * as path from "path";
+import * as xmltools from "./src/xmltools";
+
+export function generateInterface(filename: string): string {
+  const interfaceName = path.basename(
+    pascalCase(filename.substring(0, filename.lastIndexOf(".")))
+  );
+  const xml = fs.readFileSync(filename, "utf-8");
+  return xmltools.createInterface(interfaceName, xml);
+}
+
+function openTsFile(filename: string, flags: string): number {
+  filename = filename.substring(0, filename.lastIndexOf(".")) + ".ts";
+  return fs.openSync(filename, flags);
+}
+
+// checks params for flag and mutates argv
+function shouldWriteToFile(argv: string[]): boolean {
```

Comment:
I think this function is completely fine. It's easy to read an everything. :)

But, if I was writing it, I would probably just have one return point. This is more of an issue for larger functions, when it can be more difficult to get an overview if you have multiple return points.

Here is how I probably would have written this function (but you don't have to change it, unless you want to):
```typescript
function shouldWriteToFile(argv: string[]): boolean {
  const argIndex = argv.indexOf("--write-to-file");
  const hasWriteToFileFlag = (argIndex !== -1);

  if (hasWriteToFileFlag) {
    argv.splice(argIndex, 1);
  }
  
  return hasWriteToFileFlag;
}
```